homeassistant/components/hko/config_flow.py

```diff
@@ -0,0 +1,73 @@
+"""Config flow for Hong Kong Observatory integration."""
+from __future__ import annotations
+
+from asyncio import timeout
+from typing import Any
+
+from hko import HKO, LOCATIONS, HKOError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_LOCATION, Platform
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import selector
+
+from .const import API_RHRREAD, DEFAULT_LOCATION, DOMAIN, KEY_LOCATION
+
+
+def get_loc_name(item):
+    """Return an array of supported locations."""
+    return item[KEY_LOCATION]
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_LOCATION, default=DEFAULT_LOCATION): selector(
+            {
+                Platform.SELECT: {
+                    "options": list(map(get_loc_name, LOCATIONS)),
+                    "sort": True,
+                }
+            }
+        )
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hong Kong Observatory."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
```

Comment:
We don't capture the timeout exception it seems.
Probably put in a broad `Exception` capture